### PR TITLE
Adds User Onboarding and Results View Tour

### LINKED ISSUE #46: Add a user tutorial or document for how to use or app and its features - discuss on best medium.
MVP
- How do upload a file
- basic walkthrough of the areas
- filtering
- showing some conversions
  - CATT
  - Control Coverage
- Local file, S3 or Splunk

### PR DESCRIPTION
**Pull request by sesheikholeslam**
Closes #46. Implements an onboarding tour for the user which guides them through the entry interface of Heimdall. Additionally, a tour has been added to the "Results View" page of Heimdall after the user's first time loading files. 

**IMPORTANT:** Tours are only displayed once and will not be shown to the user if they have either completed, skipped, or canceled the tour. This has been done by locally storing a key in the user's cache (hopefully making this feature easily applicable to both Heimdall-Lite and Heimdall Server).

Below is a demonstration of both the onboarding and "Results View" Tour:

https://user-images.githubusercontent.com/66632401/123906266-d7b50b00-d941-11eb-9633-f44691ced310.mov

### CODE REVIEW

**Comment by rbclark** [comment]
Screen Shot 2021-06-30 at 8 38 04 AM

This breaks in lite mode since there is no database tab.

**Comment by aaronlippold** [comment]
The deploy preview on Netlify is not coming up
--------
Aaron Lippold

***@***.***

260-255-4779

twitter/aim/yahoo,etc.
'aaronlippold'

File: apps/frontend/src/components/generic/Tour.vue [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+                text: "Click here to view files loaded onto your organization's Heimdall Server",
+                buttons: defaultButtons,
+                showOn() {
+                  return document.querySelector('#select-tab-database') ? true : false;
+                }
+              },
+              {
+                attachTo: {element: '#select-tab-s3', on: 'auto'},
+                title: 'S3 Bucket',
+                text: 'Here you can access the S3 buckett',
```
Comment (camdenmoors) [inline apps/frontend/src/components/generic/Tour.vue:86]:
```suggestion
                text: 'Here you can access the S3 bucket',
```

**Comment by rbclark** [comment]
@aaronlippold the deploy preview should now be working properly.

File: apps/frontend/src/components/generic/Tour.vue [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+  <span />
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import Component from 'vue-class-component';
+import VueShepherd from 'vue-shepherd';
+import {Prop} from 'vue-property-decorator';
+import 'shepherd.js/dist/css/shepherd.css';
+
```
Comment (camdenmoors) [inline apps/frontend/src/components/generic/Tour.vue:11]:
```suggestion
import {ServerModule} from '../../store/server';
```

File: apps/frontend/src/components/generic/Tour.vue [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                text: 'Here you can easily load any supported Heimdall Data Format files',
+                buttons: defaultButtons
+              },
+              {
+                attachTo: {element: '#select-tab-database', on: 'auto'},
+                title: 'Database',
+                text: "Click here to view files loaded onto your organization's Heimdall Server",
+                buttons: defaultButtons,
+                showOn() {
+                  return document.querySelector('#select-tab-database') ? true : false;
```
Comment (camdenmoors) [inline apps/frontend/src/components/generic/Tour.vue:80]:
```suggestion
                  return ServerModule.serverMode;
```

File: apps/frontend/src/components/generic/Tour.vue [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+                attachTo: {element: '#select-tab-sample', on: 'auto'},
+                title: 'Samples',
+                text: 'Here you can use our sample files to display the power of Heimdall',
+                buttons: defaultButtons
+              },
+              {
+                attachTo: {element: '#user-icon', on: 'auto'},
+                title: 'Your Account',
+                text: 'Here you can access more user information and group information',
+                buttons: defaultButtons
```
Comment (camdenmoors) [inline apps/frontend/src/components/generic/Tour.vue:105]:
```suggestion
                buttons: defaultButtons,
                showOn() {
                  return ServerModule.serverMode;
                }
```

**Review by camdenmoors — CHANGES_REQUESTED** [review]

**Comment by camdenmoors** [comment]
Superseded by #2137

**Pull request closed without merging**